internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -520,6 +524,12 @@ func (r *BMACReconciler) reconcileSpokeBMH(ctx context.Context, bmh *bmh_v1alpha
 		return reconcileError{err}
 	}
 
+	err = r.ensureMCSCert(ctx, bmh)
```

Comment:
not sure this is the right place, we need to add before installation starts. We must ensure that we push mcs to host ignition overrides before approving host

Reply:
The code flow is changed accordingly. Thank you.